siege_utilities/geo/spatial_data.py

```
@@ -538,10 +557,21 @@ def get_year_directory_contents(
             self.cache[cache_key] = (time.time(), directories)
             return directories
 
-        except requests.exceptions.SSLError:
-            log.warning(f"SSL verification failed for year {year}, trying without verification...")
+        except requests.exceptions.SSLError as ssl_err:
+            if not _ALLOW_INSECURE_SSL:
+                log.error(
+                    "SSL verification failed for year %s; refusing to "
+                    "fall back to verify=False. Set SIEGE_INSECURE_SSL=1 "
+                    "to opt into the legacy bypass.", year,
+                )
```

Ruff (0.15.12)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)